Fixing bugs in treeinference models using hard-coded pruning

vresbok opened the pull request:
After Gizem and ViktorP's work earlier this semester, the models using hard-coded pruning in TreePPL needed to be adjusted.

They now:
- Use the corrected pickpair function.
- Have correct weighting when leaves are initiated.
- Have adjusted weighting at each "cluster" step, i.e. removing weights of previous children even when children are leaves.

This has been fixed for both JC and GTR model versions.

Also, compile and run instructions have been removed as these easily get outdates (discussed at earlier TreePPL meetings)

elegios commented on models/phylogeny/tree_inference_pruning.tppl:
```diff
@@ -117,23 +126,22 @@ function cluster(q: Matrix[Real], trees: MsgTree[], maxAge: Real, seq_len: Int)
-    let new_trees = paste0([slice(trees, 1, min), slice(trees, min + 1, max), slice(trees, max + 1, n + 1), [parent]]);
+    let min = mini(pairs[2],pairs[1]);
+    let max = maxi(pairs[2],pairs[1]);
+    let new_trees = paste0([slice(trees, 1, min), slice(trees, addi(min, 1), max), slice(trees, addi(max, 1), addi(n, 1)), [parent]]);
```
`addi(x, 1)` -> `x + 1`

elegios commented on models/phylogeny/tree_inference_pruning.tppl:
```diff
@@ -117,23 +126,22 @@ function cluster(q: Matrix[Real], trees: MsgTree[], maxAge: Real, seq_len: Int)
-    let min = pairs[2];
-    let max = pairs[1];
-    let new_trees = paste0([slice(trees, 1, min), slice(trees, min + 1, max), slice(trees, max + 1, n + 1), [parent]]);
+    let min = mini(pairs[2],pairs[1]);
+    let max = maxi(pairs[2],pairs[1]);
```
I don't think `mini` and `maxi` are exposed, so you probably need to add them. Should be just adding
```
let mathMini = mini
```
to `src/lib/standard.mc` and
```
function minInt(a : Int, b : Int) => Int {
  return mathMini(a)(b);
}
```
to `src/lib/standard.tppl`, and corresponding for `maxi`. Note the rename to `minInt`, to follow naming for `seqSumInt` and the like.

elegios reviewed (changes requested):
Sorry for the late comments. There were some changes to the standard library in #95 which unfortunately requires some changes here.

vresbok commented:
After the comments I have:
- Replaced addi and subi with + and -
- Added minInt and maxInt to standard library, adjusted model files (and removed helper functions mini and maxi)

vresbok commented:
I ran `make test` and these passed for me.
